### PR TITLE
Add linter to CI and applied some fixes

### PR DESCRIPTION
**Pull request by andresuribe87**
This PR adds the clippy linter to the repo. See https://doc.rust-lang.org/clippy/index.html for more information. 

I applied the automatic fixes. There's still one more error: https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception, which we need to decide how to handle. Open for feedback on how to best deal with it.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,14 +10,25 @@ on:
 
 env:
   CARGO_TERM_COLOR: always
+  # Make sure CI fails on all warnings, including Clippy lints
+  RUSTFLAGS: "-Dwarnings"
 
 jobs:
+  clippy_check:
```
Comment (amika-sq) [inline .github/workflows/ci.yml:17]:
```suggestion
  clippy:
```

Just `clippy` is fine.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -10,14 +10,25 @@ on:
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [ stable, nightly ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run Clippy
+        run: cargo clippy --all-targets --all-features
```
Comment (amika-sq) [inline .github/workflows/ci.yml:25]:
```suggestion
        run: cargo clippy --workspace
```

With the way our workspace is setup currently, we'll need to use the workspace flag instead.

**Review by amika-sq — COMMENTED** [review]
Looks good! Few minor comments:

> I applied the automatic fixes. There's still one more error: https://rust-lang.github.io/rust-clippy/master/index.html#/module_inception, which we need to decide how to handle. Open for feedback on how to best deal with it.

I can take this one on and fix it. We consolidated on the module naming structure here: https://github.com/TBD54566975/tbdex-rs/pull/31#discussion_r1409934783

**Review by amika-sq — APPROVED** [review]

**Comment by KendallWeihe** [comment]
cc @leordev I don't think this impacts your work, but making you aware of the use of `clippy` here

File: crates/crypto/src/key/public_key.rs [hunk 3]
```diff
@@ -39,7 +38,7 @@ mod tests {
         let signature = private_key.sign(payload).unwrap();
 
         let public_key = private_key.to_public();
-        let verification_warnings = public_key.verify(&payload, &signature).unwrap();
+        let verification_warnings = public_key.verify(payload, &signature).unwrap();
```
Comment (KendallWeihe) [inline crates/crypto/src/key/public_key.rs:41]: What provoked this change? Why is `signature` passed by reference but `payload` isn't?

**Review by KendallWeihe — APPROVED** [review]

File: crates/crypto/src/key/public_key.rs (hunk 3 above)
Comment (andresuribe87) [reply crates/crypto/src/key/public_key.rs:41]: This is the error https://rust-lang.github.io/rust-clippy/master/index.html#/needless_borrow

Comment (andresuribe87) [reply crates/crypto/src/key/public_key.rs:41]: BTW this was the result of running `cargo clippy --fix`

Comment (KendallWeihe) [reply crates/crypto/src/key/public_key.rs:41]: Oh I see, because `payload` was already initialized by reference whereas `signature` was not

**Pull request merged**